streams/src/main/java/org/apache/kafka/streams/state/internals/SessionStoreBuilder.java

```diff
@@ -52,6 +53,16 @@ private SessionStore<Bytes, byte[]> maybeWrapCaching(final SessionStore<Bytes, b
         if (!enableCaching) {
             return inner;
         }
+
+        if (!inner.persistent()) {
```

Comment:
Here's the other change I made in order to work around the current tricky situation, since in-memory stores are always "time ordered" as well, we stripe the caching if the inner store is not persistent. cc @mjsax

Reply:
If we strip the caching, this applies to eager emitting, right? -- So it would be a behavioral change? Do we want to piggy-back such a change into this KIP? Sounds "risky"?

Reply:
Sounds good, I will move this logic into the earlier stage during the topology building phase for the moment. Also cc @lihaosky who would do similar things for sliding windows.